iac-extensions/spring-config/src/main/java/org/sonar/iac/springconfig/plugin/SpringConfigSettings.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * SonarQube IaC Plugin
+ * Copyright (C) 2021-2024 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.iac.springconfig.plugin;
+
+import java.util.List;
+import org.sonar.api.PropertyType;
+import org.sonar.api.config.PropertyDefinition;
+import org.sonar.api.resources.Qualifiers;
+
+public final class SpringConfigSettings {
+  static final String ACTIVATION_KEY = "sonar.springconfig.activate";
+  static final String ACTIVATION_DEFAULT_VALUE = "true";
+  static final String FILE_PATTERNS_KEY = "sonar.springconfig.file.patterns";
+  static final String DEFAULT_FILE_PATTERNS = "**/src/main/resources/**/application*.properties," +
+    "**/src/main/resources/**/application*.yaml,**/src/main/resources/**/application*.yml";
+  private static final String JAVA_CATEGORY = "Java";
+  private static final String GENERAL_SUBCATEGORY = "General";
+
+  private SpringConfigSettings() {
+  }
+
+  public static List<PropertyDefinition> getGeneralProperties() {
+    return List.of(
+      PropertyDefinition.builder(ACTIVATION_KEY)
+        .index(1)
```

Comment:
Here we should not use `.index(1)` It causes that in SQ in Project Settings page -> Language -> Java this setting is displayed as first, before `sonar.java.file.suffixes` what is preferable on the top of the settings list. Unfortunatly sonar-java doesn't set any indexes :-( 
For sure we should either remove it, or increase to index or contact JVM squad and discuss it.

Reply:
Here, we should be careful and double-check the UI as we should add configuration to the Java language.

Reply:
Yes, good catch, I updated the PR to have it in a separate section
![image](https://github.com/SonarSource/sonar-iac/assets/23018090/4bcc00cb-7f7c-4836-9a49-c40802d431ad)